packages/@aws-cdk/pipelines/lib/stage.ts

```diff
@@ -76,14 +81,79 @@ export class CdkStage extends CoreConstruct {
    */
   public addApplication(appStage: Stage, options: AddStageOptions = {}) {
     const asm = appStage.synth({ validateOnSynthesis: true });
-    const extraRunOrderSpace = options.extraRunOrderSpace ?? 0;
+    const extraRunOrderSpace = options.extraRunOrderSpace ?? 0 + (options.securityCheck ? 2 : 0);
 
     if (asm.stacks.length === 0) {
       // If we don't check here, a more puzzling "stage contains no actions"
       // error will be thrown come deployment time.
       throw new Error(`The given Stage construct ('${appStage.node.path}') should contain at least one Stack`);
     }
 
+    if (options.securityCheck) {
+      const preApproveLambda = new lambda.Function(this, `${this.stageName}AutoApprove`, {
+        functionName: 'AutoApproveLambda',
```

Comment:
Don't give this Lambda a physical name. 

Just stick `${preApproveLambda.functionName}` into the command string, everything will work out great!